### PR TITLE
DOCSP-43050: Repo cleanup for standardization + Index page

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43050>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/c/DOCSP-43050-index/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -1,118 +1,113 @@
...
+.. asking questions and receiving help in the :ref:`Issues & Help <c-issues-and-help>` section.
 
-.. include:: /libmongoc/includes/mongodb-compatibility-table-c.rst
+.. TODO
+.. Compatibility
+.. -------------
 
-.. include:: /libmongoc/includes/older-server-versions-unsupported.rst
+.. For the compatibility charts that show the recommended {+driver-short+} version for each
+.. MongoDB Server version, see the :ref:`Compatibility <c-compatibility>` section.
```
Comment (norareidy) [inline source/index.txt:96]: S: [nit] use the {+mdb-server+} constant
```suggestion
.. {+mdb-server+} version, see the :ref:`Compatibility <c-compatibility>` section.
```

File: source/index.txt [hunk 2]
```diff
@@ -1,118 +1,113 @@
...
-Welcome to the documentation site for the official MongoDB C driver.
-You can add the driver to your application to work with MongoDB in C.
-Download the required libraries, ``libmongoc`` and ``libbson``, by following
-the :ref:`obtaining_libraries` instructions or set up a runnable project by
-following our tutorial.
-
-- :ref:`Tutorial <mongoc_tutorial>`
-
-- :ref:`Usage Guide <libmongoc>`
+Welcome to the documentation site for the {+driver-long+}, also known as **libmongoc**,
```
Comment (norareidy) [inline source/index.txt:15]: Q: are we just documenting libmongoc?

**Review by norareidy — APPROVED** [review]
Approving, but I'll follow up about the question I left

File: source/index.txt (hunk 2 above)
Comment (mcmorisi) [reply source/index.txt:15]: Answered offline – summary is we plan to include `libbson` info in the proper pages in the new standardized ToC (Installation instructions, BSON guide)

**Pull request merged**